test/generate.test.ts

```diff
@@ -133,6 +152,53 @@ describe('generateClassAssignment ', () => {
     await assembly.cleanup();
   });
 
+  test('returns undefined when only protected/private methods present', async () => {
+    const assembly = await AssemblyFixture.fromSource(
+      {
+        'index.ts': `
+        export class ClassA {
+          protected static protectedMethod() { new ClassA(); }
+          private constructor() {}
+        }`,
+      },
+      {
+        name: 'my_assembly',
+        jsii: DUMMY_ASSEMBLY_TARGETS,
+      },
+    );
+
+    const ts = new reflect.TypeSystem();
+    await ts.load(assembly.directory);
+
+    const type = ts.findClass('my_assembly.ClassA');
+    expect(generateAssignmentStatement(type)).toBeUndefined();
+
+    await assembly.cleanup();
+  });
+
+  test('returns undefined for abstract classes', async () => {
```

Comment:
this is just an extra test that should be nice to have. verifies that this has been true